lib/auth/auth.js

```diff
@@ -117,11 +116,12 @@ auth.generateV4Headers =
         request.setHeader('host', request._headers.host);
         request.setHeader('x-amz-date', amzDate);
         request.setHeader('x-amz-content-sha256', payloadChecksum);
-        Object.assign(request.headers, { host: request.getHeader('host') });
-        Object.assign(request.headers, { 'x-amz-date': amzDate });
-        Object.assign(request.headers,
-                      { 'x-amz-content-sha256': payloadChecksum });
-
+        Object.assign(request.headers, request._headers);
+        const signedHeaders = Object.keys(request._headers)
+            .filter(headerName =>
+            headerName.startsWith('x-amz-') || headerName.startsWith('x-scal')
```

Comment:
I include both in the required headers test in case x-scal headers are created later.